front/context/src/AccountsContext.tsx

```diff
@@ -2,59 +2,71 @@
 // This software may be modified and distributed under the terms
 // of the Apache-2.0 license. See the LICENSE file for details.
 
-import React, { createContext, useEffect, useState } from 'react';
+import { web3Accounts, web3Enable } from '@polkadot/extension-dapp';
+import {
+  InjectedAccountWithMeta,
+  InjectedExtension,
+} from '@polkadot/extension-inject/types';
+import React, { createContext, useState } from 'react';
 
-import { InjectedAccountExt } from './types';
+interface AccountsContext {
+  accounts: InjectedAccountWithMeta[];
+  readonly extension: InjectedExtension;
+  fetchAccounts: () => Promise<void>;
+  isExtensionReady: boolean;
+}
 
-export const AccountsContext = createContext({
-  injectedAccounts: [] as InjectedAccountExt[],
-});
+export const AccountsContext = createContext({} as AccountsContext);
 
 interface Props {
-  children: React.ReactNode;
+  children: React.ReactElement;
+  /**
+   * Slug specific to your app,
+   * @see https://github.com/polkadot-js/extension/tree/master/packages/extension-inject#polkadotextension-inject
+   */
+  originName: string;
 }
 
 export function AccountsContextProvider(props: Props): React.ReactElement {
-  const { children } = props;
-  const [injectedAccounts, setInjected] = useState<InjectedAccountExt[]>(
-    [] as InjectedAccountExt[]
-  );
+  const { children, originName } = props;
+  const [accounts, setAccounts] = useState<InjectedAccountWithMeta[]>([]);
+  const [extension, setExtension] = useState<InjectedExtension>();
+  const [isReady, setIsReady] = useState(false);
+
+  /**
+   * Fetch accounts from the extension
+   */
+  async function fetchAccounts(): Promise<void> {
+    const extensions = await web3Enable(originName);
 
-  useEffect(() => {
-    const getInjected = async (): Promise<void> => {
-      if (typeof window !== 'undefined') {
-        const {
-          web3Accounts,
-          web3Enable,
-          // Can we do `import { } from '@polkadot/extension-dapp'` on the top?
-          // eslint-disable-next-line @typescript-eslint/no-var-requires
-        } = require('@polkadot/extension-dapp');
-        await web3Enable('nomidot');
-        const [injectedAccounts] = await Promise.all([
-          web3Accounts().then((accounts: InjectedAccountExt[]) =>
-            accounts.map(
-              ({ address, meta }): InjectedAccountExt => ({
-                address,
-                meta: {
-                  ...meta,
-                  name: `${meta.name} (${
-                    meta.source === 'polkadot-js' ? 'extension' : meta.source
-                  })`,
-                },
-              })
-            )
-          ),
-        ]);
-
-        setInjected(injectedAccounts);
-      }
-    };
-
-    getInjected();
-  }, []);
+    if (!extensions.length) {
+      throw new Error(
+        'No extension found. Please install PolkadotJS extension.'
+      );
+    }
+
+    setExtension(extensions[0]);
```

Comment:
I'm fine with this for now before incorporate @goldsteinsveta 's designs for switching between accounts.